main.js

```diff
@@ -16,108 +16,145 @@ let bootstrapFixtures = {
-              "sourceCodeLink":"https://github.com/perspect3vism/direct-message-language"
-            },
-            "proof":{
-              "signature":"d5f120f0cd225386499c54addd0bd9e5b0706c448d6211c2cf94333f8c78734612f8a3606e8e188ffb370fca6bd6ae301337384b24809febb1d12c38c6cdebcf",
-              "key":"#zQ3shkkuZLvqeFgHdgZgFMUx8VGkgVWsLA83w2oekhZxoCW2n",
-              "valid":true
-            }
-          },
-          bundle: fs.readFileSync(path.join('src', 'languages', 'direct-message-language', 'build', 'bundle.js')).toString()
-        }
+            bundle: fs.readFileSync(path.join('src', 'languages', 'direct-message-language', 'build', 'bundle.js')).toString()
+          }
+        ],
+        perspectives: [{
+          address: '__world',
+          expression: bootstrapFixtures.worldPerspective
+        }]
+      },
+      appBuiltInLangs: [
+        "social-context",
+        "note-ipfs",
+        "virtual-icons"
       ],
-      perspectives: [{
-        address: '__world',
-        expression: bootstrapFixtures.worldPerspective
-      }]
-    },
-    appBuiltInLangs: [
-      "social-context",
-      "note-ipfs",
-      "virtual-icons"
-    ],
-    mocks: false
-  })
-  .then((core) => {
-    ad4mCore = core
-
-    console.log(
-      "\x1b[36m%s\x1b[0m",
-      "Starting account creation splash screen"
-    );
-
-    const splash = createSplash()
-    ad4mCore.waitForAgent().then(async () => {
+      mocks: false
+    })
+    .then((core) => {
+      ad4mCore = core
+  
       console.log(
         "\x1b[36m%s\x1b[0m",
-        "Agent has been init'd. Controllers now starting init..."
+        "Starting account creation splash screen"
       );
-      ad4mCore.initControllers();
-      console.log("\x1b[32m", "Controllers init complete!");
-
-      console.log(
-        "\x1b[36m%s\x1b[0m",
-        "Initializing languages..."
-      );
-      await ad4mCore.initLanguages()
-      console.log("\x1b[32m", "All languages initialized!");
-
-
-      createWindow()
-      splash.close()
+  
+      const splash = createSplash()
+      ipcMain.on('port-request', (event, arg) => {
+        event.returnValue = executorPort
+      })
+      ad4mCore.waitForAgent().then(async () => {
+        console.log(
+          "\x1b[36m%s\x1b[0m",
+          "Agent has been init'd. Controllers now starting init..."
+        );
+        ad4mCore.initControllers();
+        console.log("\x1b[32m", "Controllers init complete!");
+  
+        console.log(
+          "\x1b[36m%s\x1b[0m",
+          "Initializing languages..."
+        );
+        await ad4mCore.initLanguages()
+        console.log("\x1b[32m", "All languages initialized!");
+  
+  
+        createWindow()
+        splash.close()
+      });
     });
-  });
-
+  }
+  else {
+    try {
+      executorPort = fs.readFileSync(path.join(app.getPath("appData"), '../.local/share/ad4m/executor-port'), {encoding:'utf8', flag:'r'})
```

Comment:
You could use this package https://github.com/perspect3vism/appdata-path to get the path, would work on all platforms.